soak/bootstrap.sh

```diff
@@ -0,0 +1,204 @@
+# Release name of kube-prometheus helm chart
+PROM_CHART_RELEASE_NAME="kube-prom"
+# Namespace for the prometheus helm chart
+PROM_NAMESPACE="prometheus"
+# Local port to access Prometheus dashbaord
+LOCAL_PROMETHEUS_PORT=9090
+# Local port to access Prometheus dashbaord
+LOCAL_GRAFANA_PORT=3000
+
+### SOAK TEST RUNNER CONFIGURATION ###
+# The binary used to build and push the container images
+OCI_BUILDER=${OCI_BUILDER:-"docker"}
+# The public ECR repository URI where your soak test runner image will be stored.
+DEFAULT_SOAK_IMAGE_REPO_NAME="ack-$AWS_SERVICE-soak"
+SOAK_IMAGE_REPO_NAME=${SOAK_IMAGE_REPO_NAME:-$DEFAULT_SOAK_IMAGE_REPO_NAME}
+# Image tag for soak-test-runner image.
+SOAK_IMAGE_TAG="v0.0.1"
+# Platform for soak-test-runner image.
+SOAK_IMAGE_PLATFORM="linux/amd64"
+# Release name of soak-test-runner helm chart.
+SOAK_RUNNER_CHART_RELEASE_NAME="soak-test-runner"
+
+# Total test duration is calculated as sum of TEST_DURATION_MINUTES,
+# TEST_DURATION_HOURS and TEST_DURATION_DAYS after converting them in
+# minutes. Override following variables accordingly to set your soak test
+# duration. Default value: 24 hrs.
+TEST_DURATION_DAYS=${TEST_DURATION_DAYS:-1}
+TEST_DURATION_HOURS=${TEST_DURATION_HOURS:-0}
+TEST_DURATION_MINUTES=${TEST_DURATION_MINUTES:-0}
+NET_SOAK_TEST_DURATION_MINUTES=$(($TEST_DURATION_MINUTES + $TEST_DURATION_HOURS*60 + $TEST_DURATION_DAYS*24*60))
+
+export HELM_EXPERIMENTAL_OCI=1
+
+AWS_CLI="aws --region $DEPLOY_REGION"
+AWS_ECR_PUBLIC_CLI="aws --region us-east-1 ecr-public"
+
+[ ! -d "$CONTROLLER_DIR" ] && { >&2 echo "Error: Service controller directory does not exist: $CONTROLLER_DIR"; exit 1; }
+
+# Check and create the public ECR repository
+if $AWS_ECR_PUBLIC_CLI describe-repositories --repository-name $SOAK_IMAGE_REPO_NAME > /dev/null 2>&1; then
+    echo "ECR public repository already exists"
+else
+    echo -n "Creating ECR public repository ... "
+    $AWS_ECR_PUBLIC_CLI create-repository --repository-name $SOAK_IMAGE_REPO_NAME
+    echo "ok."
+fi
+
+# Check and create the EKS cluster
+CLUSTER_NAME=$(yq eval -e ".metadata.name" $CLUSTER_CONFIG_PATH 2> /dev/null) || CLUSTER_NAME=$DEFAULT_CLUSTER_NAME
+if $AWS_CLI eks describe-cluster --name $CLUSTER_NAME > /dev/null 2>&1; then
+    echo "EKS cluster already exists"
+else
+    echo "Creating EKS cluster ... "
+    eksctl create cluster -f $CLUSTER_CONFIG_PATH
+    echo "ok."
+fi
+
+$AWS_CLI eks update-kubeconfig --name $CLUSTER_NAME > /dev/null
+
+# Install the controller into the cluster
+CONTROLLER_SERVICE_ACCOUNT_NAME=$(yq eval -e ".iam.serviceAccounts[0].metadata.name" $CLUSTER_CONFIG_PATH 2> /dev/null) || \
+    { >&2 echo "Error: IRSA is not included in this cluster config. IRSA is required for soak testing"; exit 1; }
+
+if helm list -n $CONTROLLER_NAMESPACE 2> /dev/null | grep -q $CONTROLLER_CHART_RELEASE_NAME; then
+    echo "Controller Helm release ($CONTROLLER_CHART_RELEASE_NAME) already installed in cluster"
```

Comment:
shouldn't the installation be upgraded with latest controller image when it already exists?

Reply:
Yeah. When I initially designed this script, I was thinking about how it works only for the first installation. I guess I should expand it to be used for any future releases, as well. I'll update this to match the soak test runner.